### PR TITLE
 Load config outside doorman (fixes #68) 

### LINKED ISSUE #68: Move policies loading/downloading out of doorman package
From #67 : 

> Also I think the config should handle loading and parsing of the sources. Then the configuration passed to doorman.New(doormanConfig) would already contain all the policy data it needs. This is a larger refactor.

### PR DESCRIPTION
**Pull request by leplatrem**
Fixes #68

### CODE REVIEW

**Comment by leplatrem** [comment]
The diff is pretty huge, and I don't expect thorough review (coverage and tests back me here).

@mostlygeek could you please tell me what do you think of the current project layout (`settings.go`, `config` package, etc.)? Thanks!

File: main.go [hunk 1]
```diff
@@ -4,27 +4,45 @@ import (
...
 )
 
+func init() {
+	config.AddLoader(&config.FileLoader{})
+	config.AddLoader(&config.GithubLoader{
+		Token: settings.GithubToken,
+	})
+}
+
 func setupRouter() (*gin.Engine, error) {
```
Comment (mostlygeek) [inline main.go]: `setupRouter()` doesn't seem necessary as it's only used in `main()`.  It's probably fine to just put all this logic right into `main()`.  This isn't a big deal.

File: main.go [hunk 2]
```diff
@@ -4,27 +4,45 @@ import (
...
+	setupLogging()
+
 	r.Use(HTTPLoggerMiddleware())
 
 	// Setup doorman and load configuration files.
-	w := doorman.NewDefaultLadon(doorman.Config{
-		Sources: config.Sources,
-	})
-	if err := w.LoadPolicies(); err != nil {
+	d := doorman.NewDefaultLadon()
```
Comment (mostlygeek) [inline main.go]: This can be done after attempting config load. Might as well put all the `doorman` bits together.

File: main.go (hunk 1 above)
Comment (mostlygeek) [reply main.go]: oh i see you use `setupRouter()` so you can test that it works. NM.

**Comment by mostlygeek** [comment]
Everything looks great. My preference is to export `http.Handler` instead of passing `*gin.Engine` around. However, it's really easy to follow the code and figure out where routes get added to gin and how things are wired together. 

I think this is a much better layout / design. R+ merge away

**Pull request merged**